src/testing/mock.rs

```diff
@@ -372,21 +310,20 @@ impl Future for NetworkHub {
         for peer_id in disconnected_peers {
             self.peers.lock().remove(&peer_id);
         }
-        for addr_msg in buffer {
+        for (data, node_id) in buffer {
             let rand_sample = rand::random::<f64>();
             if rand_sample > self.reliability {
                 debug!("Simulated network fail.");
                 continue;
             }
             let mut peers = self.peers.lock();
-            if let Some(peer) = peers.get_mut(&addr_msg.receiver.clone()) {
+            if let Some(peer) = peers.get_mut(&node_id) {
                 let hooks = self.hook_list.lock();
                 for hook in hooks.iter() {
-                    hook.update_state(addr_msg.clone());
+                    hook.update_state(data.clone());
```

Comment:
It would be good to allow the hook to see not only the data, but the sender and receiver as well. This then allows to make sophisticated tests without the need to modify this Router mock.

Reply:
Done, this also somewhat improved the semantics of incoming messages, although this isn't used in any way.